### PR TITLE
Feature/ui hydrator required fields count fix

### PR DESCRIPTION
**Pull request by elfenheart**
- Fix issue with required fields count not appearing on the Node
- Fix issue with asterisk not appearing on NoConfig template

### CODE REVIEW

File: cdap-ui/app/features/hydrator/services/pipeline-error-factory.js [hunk 1]
```diff
@@ -166,31 +165,19 @@ angular.module(PKG.name + '.feature.hydrator')
...
-            for (i=0; i< keys.length; i++) {
-              var property = plugin.properties[keys[i]];
-              if (plugin._backendProperties[keys[i]] && plugin._backendProperties[keys[i]].required && (!property || property === '')) {
-                requiredFieldCount += 1;
-              }
+      if (angular.isObject(node._backendProperties) && Object.keys(node._backendProperties).length) {
+
+        angular.forEach(node._backendProperties, function (value, key) {
+          if (value.required) {
+            if (!node.plugin.properties || !node.plugin.properties[key] || node.plugin.properties[key] === '') {
```
Comment (anwar6953) [inline cdap-ui/app/features/hydrator/services/pipeline-error-factory.js]: the third condition of this isn't required, right? (its implicit in the second?)

**Comment by ajainarayanan** [comment]
:+1: LGTM

File: cdap-ui/app/features/hydrator/services/pipeline-error-factory.js (hunk 1 above)
Comment (elfenheart) [inline cdap-ui/app/features/hydrator/services/pipeline-error-factory.js]: Sadly no, javascript treats empty string as boolean true..

Comment (anwar6953) [inline cdap-ui/app/features/hydrator/services/pipeline-error-factory.js]: not a big deal

**Comment by anwar6953** [comment]
lgtm

**Pull request merged**